File: app/src/main/java/com/mml/movies/network/ApiService.java
package com.mml.movies.network;

import com.mml.movies.network.models.ApiResult;

import java.util.Map;
import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface ApiService {
    @GET("movie/popular")
    Single<ApiResult> getTrending(@QueryMap Map<String, String> queries);

    @GET("movie/now_playing")
    Single<ApiResult> getNowPlaying(@QueryMap Map<String, String> queries);

    @GET("search/movie")
    Single<ApiResult> getSearchResult(@QueryMap Map<String, String> queries);
}

File: app/src/main/java/com/mml/movies/network/ApiService.kt
package com.mml.movies.network

import com.mml.movies.network.models.ApiResult
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.core.Single
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface ApiService {

    @GET("movie/popular")
    fun getTrending(@QueryMap queries:Map<String, String>): Single<ApiResult>

    @GET("movie/now_playing")
    fun getNowPlaying(@QueryMap queries:Map<String, String>): Single<ApiResult>

    @GET("search/movie")
    fun getSearchResult(@QueryMap queries:Map<String, String>): Single<ApiResult>
}